### PR TITLE
GOZER #45 - handle varying segment delimiters

### PR DESCRIPTION
**Pull request by CodeSmell**
Handle various segment delimiters when message has no line breaks per issue #45

### CODE REVIEW

File: src/main/java/com/walmartlabs/x12/X12Parser.java [hunk 1]
```diff
@@ -53,9 +64,28 @@
             return segments;
         } else {
             // if there is only one line in the source data
-            // we should attempt to use the segment separator passed in
-            // and see if we can split up this source data
-            return splitSourceDataIntoSegments(sourceData, "\\" + DEFAULT_SEGMENT_SEPARATOR);
+            // we should attempt to use the default segment separator
+            segments = splitSourceDataIntoSegments(sourceData, "\\" + DEFAULT_SEGMENT_SEPARATOR);
+            if (segments != null && segments.size() > 1) {
```
Comment (bhatms) [inline src/main/java/com/walmartlabs/x12/X12Parser.java]: can we use
`segments != null && !segments.isEmpty()` or
`CollectionUtils.isEmpty(segments)`

File: src/main/java/com/walmartlabs/x12/X12Parser.java [hunk 2]
```diff
@@ -53,9 +64,28 @@
...
+                return segments;
+            } else {
+                // if there is only one line in the source data
+                // now try to get the specified delimiter
+                return splitSourceDataIntoSegments(sourceData, "\\" + findSegmentDelimiterCharacter(sourceData));
```
Comment (bhatms) [inline src/main/java/com/walmartlabs/x12/X12Parser.java]: can we create a String variable first
`String segmentSeparatorRegEx = "\\" + findSegmentDelimiterCharacter(sourceData))`
and then pass it to this call
`splitSourceDataIntoSegments (.. , ..)` easy to read

File: src/test/java/com/walmartlabs/x12/standard/StandardX12ParserTest.java [hunk 3]
```diff
@@ -176,6 +176,28 @@ public void test_Parsing_BaseDocument_empty_lines_at_end() throws IOException {
...
+    
+    @Test
+    public void test_Parsing_BaseDocument_no_line_breaks_short_file() {
+        try {
+            String sourceData = "ISA*01*0000000000*01*0000000000*ZZ*ABCDEFGHIJKLMNO*ZZ";
+            StandardX12Parser localParser = this.createParserWithRegistrationViaCollection();
+            localParser.parse(sourceData);
+            fail("expected parsing exception");
+        } catch (X12ParserException e) {
+            e.printStackTrace();
```
Comment (bhatms) [inline src/test/java/com/walmartlabs/x12/standard/StandardX12ParserTest.java]: is printStackTrace needed ?

Comment (CodeSmell) [reply src/test/java/com/walmartlabs/x12/standard/StandardX12ParserTest.java]: left over from debugging

File: src/main/java/com/walmartlabs/x12/X12Parser.java [hunk 4]
```diff
@@ -53,9 +64,28 @@
...
+    /**
+     * get the segment delimiter/separator character
+     * @param sourceData
+     * @return the character at the 106th position
+     */
+    default Character findSegmentDelimiterCharacter(String sourceData) {
+        if (sourceData != null && sourceData.length() > SEGMENT_SEPARATOR_INDEX) {
+            return Character.valueOf(sourceData.charAt(SEGMENT_SEPARATOR_INDEX));
+        } else {
+            return null;
```
Comment (patelch) [inline src/main/java/com/walmartlabs/x12/X12Parser.java]: What happens if this method returns null when you call it from here:
`splitSourceDataIntoSegments(sourceData, "\\" + findSegmentDelimiterCharacter(sourceData));`

Comment (CodeSmell) [reply src/main/java/com/walmartlabs/x12/X12Parser.java]: throws `PatternSyntaxException`, I added that to JavaDocs & added a test
good catch

**Pull request merged**